unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -1671,7 +1671,9 @@ abstract public class BaseFPSAgentController : MonoBehaviour {
             // in the multiagent setting, explicitly giving this information for now.
             objMeta.visible = isVisible; // simObj.isVisible;
 
-            objMeta.obstructedByTransparentObject = !simObj.isInteractable;// object is not interactable. It may still be visible, but occluded by some transparent object (glass)
+            //determines if the objects is unobstructed and interactable. Objects visible behind see-through geometry like glass will be isInteractable=False even if visible
+            //note using forceAction=True will ignore the isInteractable requirement
+            objMeta.isInteractable = simObj.isInteractable;
```

Comment:
The issue here is that a `SimObjPhysics` should not "know" if it is interactable. We should do the same thing we do for visibility, i.e. when generating the object metadata we run
```c#
HashSet<SimObjPhysics> visibleSimObjsHash = new HashSet<SimObjPhysics>(GetAllVisibleSimObjPhysics(...));
```
with something like 
```
HashSet<SimObjPhysics> visibleSimObjsHash = new HashSet<SimObjPhysics>(GetAllVisibleSimObjPhysics(...));
HashSet<SimObjPhysics> interactableSimObjsHash = new HashSet<SimObjPhysics>(GetAllInteractableSimObjPhysics(...));
```
Of course it would be more efficient to have the two above function calls combined into one.

Reply:
The two options I see are to consider moving to the distance based visibility scheme introduced last year, but not necessarily compatible with the current default scheme.  Then this method can be used in `getTargetObject`:
https://github.com/allenai/ai2thor/blob/fb126c24d7682bafcbd5192fd347408ed893f2bb/unity/Assets/Scripts/BaseFPSAgentController.cs#L2744

Alternatively, we could create a new method based on this method:
https://github.com/allenai/ai2thor/blob/fb126c24d7682bafcbd5192fd347408ed893f2bb/unity/Assets/Scripts/BaseFPSAgentController.cs#L2856
that is able to just tell you whether a single SimObjPhysics is visible.

Reply:
Sorry for all the back and forth, I read through the code once again - and this and the getTargetObject below don't actually have any problems with multi-agent usage since a visibility check is performed immediately before consulting the `isInteractable` flag.  That said,  I agree with @Lucaweihs's point about `isInteractable` should not reside on the SimObjPhysics since its a boolean that reflects that state of both the agent and the object (not just the object).  I have an idea for how to refactor this - I can update this PR with the change.